File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/PreviousProfile.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Entity
@Table(name = "PREVIOUS_PROFILE")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class PreviousProfile {
    @Id
    @Length(max = 10)
    private String offenderNo;

    private String escape;
    private String extremism;
    private String soc;
    private String violence;

    @NotNull
    private LocalDateTime executeDateTime;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/PreviousProfile.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.validation.constraints.NotNull
import org.hibernate.validator.constraints.Length
import java.time.LocalDateTime

@Entity
@Table(name = "PREVIOUS_PROFILE")
data class PreviousProfile(
  @Id
  @Length(max = 10)
  val offenderNo: String,
  var escape: String,
  var soc: String,
  var violence: String,
  var executeDateTime: @NotNull LocalDateTime,
)
